### PR TITLE
Phase2-hgx332 Add debug tools to investigate shortcomings of V16/V17 geometries of HGCal

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Add debug tools to investigate shortcomings of V16/V17 geometries of HGCal

#### PR validation:

Use the tests to find out issues in HGCal geometry versions V16 and V17

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by AdrianoDee** [comment]
please abort
(@bsunanda the latest IB is broken, as soon as #40206 will get in a new one will be built)

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

**Comment by emanueleusai** [comment]
+1

**Comment by emanueleusai** [comment]
type hgcal

**Comment by bsunanda** [comment]
@srimanob Please sign this

File: SimG4Core/Application/python/g4SimHits_cfi.py [hunk 1]
```diff
@@ -519,7 +519,7 @@
         WaferAngles      = cms.untracked.vdouble(90.0,30.0),
         WaferSize        = cms.untracked.double(123.7),
         MouseBite        = cms.untracked.double(2.5),
-        CheckID          = cms.untracked.bool(True),
```
Comment (srimanob) [inline SimG4Core/Application/python/g4SimHits_cfi.py:522]: What is an implication on turning this CheckID to off?

Comment (bsunanda) [reply SimG4Core/Application/python/g4SimHits_cfi.py:522]: This parameter was not used earlier. Now it is used to do some debug testing which is expensive in CPU time. So by default, it is better to set it as False and when checks are needed set it to True. So this change has no real effect on standard production

**Comment by srimanob** [comment]
+Upgrade

This PR adds a debug tool for HGCAL geometries. No change is expected. The change in g4SimHits should not effect simulation as stated in  https://github.com/cms-sw/cmssw/pull/40168#discussion_r1038150722

**Comment by perrotta** [comment]
+1

**Pull request merged**